tests/compiler.js

```diff
@@ -9,7 +9,7 @@ const diff = require("./util/diff");
 const asc = require("../cli/asc.js");
 const rtrace = require("../lib/rtrace");
 const cluster = require("cluster");
-const numCPUs = require('os').cpus().length;
+const coreCount = require("physical-cpu-count");
```

Comment:
How about this simple routine:
```js
const os = require('os');

const coreCount = os.cpus().filter((cpu, i) => {
  const hyperthreading = cpu.model.includes('Intel')
  return !hyperthreading || i & 1
}).length;
```
instead extra dependencies

Reply:
That doesn't seem sufficient. For example, a 9900K has 8 cores and 16 threads while a 9700K has 8 cores and 8 threads. Same happens for AMD where some models have SMT (their version of hyperthreading) while others don't.

Reply:
Yeah. You are right